redbot/cogs/mutes/mutes.py

```diff
@@ -0,0 +1,1529 @@
+                    guild, channel, author, user, audit_reason
+                )
+
+                if success["success"]:
+                    success_list.append(user)
+                    await modlog.create_case(
+                        self.bot,
+                        guild,
+                        ctx.message.created_at.replace(tzinfo=timezone.utc),
+                        "cunmute",
+                        user,
+                        author,
+                        reason,
+                        until=None,
+                        channel=channel,
+                    )
+                else:
+                    issue_list.append((user, success["reason"]))
+        if success_list:
+            if channel.id in self._channel_mutes and self._channel_mutes[channel.id]:
+                await self.config.channel(channel).muted_users.set(self._channel_mutes[channel.id])
+            else:
+                await self.config.channel(channel).muted_users.clear()
+            await ctx.send(
+                _("{users} unmuted in this channel.").format(
+                    users=humanize_list([f"{u}" for u in success_list])
+                )
+            )
+        if issue_list:
+            msg = _("The following users could not be unmuted\n")
+            for user, issue in issue_list:
+                msg += f"{user}: {issue}\n"
+            await ctx.send_interactive(pagify(msg))
+
+    async def mute_user(
+        self,
+        guild: discord.Guild,
+        author: discord.Member,
+        user: discord.Member,
+        until: Optional[datetime] = None,
+        reason: Optional[str] = None,
+    ) -> Dict[
+        str, Optional[Union[List[Tuple[discord.abc.GuildChannel, str]], discord.Member, bool, str]]
+    ]:
+        """
+        Handles muting users
+        """
+        permissions = user.guild_permissions
+        ret: Dict[
+            str,
+            Union[bool, Optional[str], List[Tuple[discord.abc.GuildChannel, str]], discord.Member],
+        ] = {
+            "success": False,
+            "reason": None,
+            "channels": [],
+            "user": user,
+        }
+        # TODO: This typing is ugly and should probably be an object on its own
+        # along with this entire method and some othe refactorization
+        # v1.0.0 is meant to look ugly right :')
+        if permissions.administrator:
+            ret["reason"] = _(MUTE_UNMUTE_ISSUES["is_admin"])
+            return ret
+        mute_role = await self.config.guild(guild).mute_role()
+
+        if mute_role:
+            if not await self.is_allowed_by_hierarchy(guild, author, user):
+                ret["reason"] = _(MUTE_UNMUTE_ISSUES["hierarchy_problem"])
+                return ret
+            role = guild.get_role(mute_role)
+            if not role:
+                ret["reason"] = _(MUTE_UNMUTE_ISSUES["role_missing"])
+                return ret
+            if not guild.me.guild_permissions.manage_roles:
```

Comment:
As said in the previous review comment
> The same goes for checking whether the role we're trying to assign can be assigned by the bot by checking its position against the bot's top role.

If you want to, you can split this into 2 errors, but I won't require that.
```suggestion
            if (
                not guild.me.guild_permissions.manage_roles
                or role >= guild.me.top_role
            ):
```